File: tests/go/cases/case_242_thread_pool_capture_04.go
package escape_tests

var retainedCase242 = []map[string]string{}

func Case242ThreadPoolCapture04(input string) string {
	raw := input
	if raw == "" {
		raw = "sample"
	}
	payload := map[string]string{
		"task": "thread_pool_capture_04",
		"entity": "extreme",
		"stage": "stress",
		"input": raw,
		"checksum": "thread_pool_capture_04:" + raw,
	}
	// ESCAPE: goroutine captures payload and retains it.
	go func(v map[string]string) { retainedCase242 = append(retainedCase242, v) }(payload)
	return "ok"
}

File: tests/java/src/main/java/com/escape/tests/cases/Case242ThreadPoolCapture04.java
package com.escape.tests.cases;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.function.Supplier;

/** Task case 242: thread_pool_capture_04 deep stress pattern. */
public class Case242ThreadPoolCapture04 {
    private static final List<Map<String, String>> RETAINED_AUDIT = new ArrayList<>();
    private static final List<Supplier<String>> RETAINED_HANDLERS = new ArrayList<>();

    private interface Sink {
        void put(Map<String, String> obj);
    }

    private static void sink(Map<String, String> obj) {
        RETAINED_AUDIT.add(obj);
    }

    public static String execute(String input) {
        String taskName = "thread_pool_capture_04";
        String raw = (input == null || input.isEmpty()) ? "sample" : input;
        Map<String, String> payload = new HashMap<>();
        payload.put("task", taskName);
        payload.put("entity", "extreme");
        payload.put("stage", "stress");
        payload.put("input", raw);
        payload.put("checksum", taskName + ":" + raw.length());
        // ESCAPE: background thread captures payload.
        new Thread(() -> RETAINED_AUDIT.add(payload)).start();
        return "ok";
    }
}
